bin/ch-test

```diff
@@ -186,18 +337,35 @@ fi
 
 # Ensure BATS_TMPDIR is set to /tmp (issue #278).
 if [[ -n $BATS_TMPDIR ]]; then
-    if [[ ! $BATS_TMPDIR == '/tmp/'* ]]; then
+    if [[ "$BATS_TMPDIR" != '/tmp'* ]]; then
         fatal "BATS_TMPDIR: must be /tmp but found $BATS_TMPDIR (issue #278)"
     fi
 fi
 
+# Create a directory in user space to hold auto-generated test artifacts
+tmpdir=${HOME}/.charliecloud/ch-test.tmp.$USER
+if [[ ! -d $tmpdir ]]; then
+    mkdir -p "$tmpdir"
+    chmod 700 "$tmpdir"
+    chmod 700 "$(dirname "$tmpdir")"
+fi
+
+if [[ $(stat -c %a "$tmpdir") != '700' ]]; then
+    fatal "BATS_TMPDIR: $tmpdir doesn't have octal perms 700"
+fi
+
+if [[ $(stat -c %a "$(dirname "$tmpdir")") != '700' ]]; then
+    fatal "BATS_TMPDIR: $(dirname "$tmpdir") doesn't have  octal perms 700"
+fi
```

Comment:
I think you do not need to check this for the case where you created the directory. But, can you reduce the number of code paths to always create it?